src/librustc_typeck/check/method/probe.rs

```diff
@@ -758,9 +758,8 @@ impl<'a, 'gcx, 'tcx> ProbeContext<'a, 'gcx, 'tcx> {
                         self.span, infer::FnCall, &fty);
 
                     if let Some(self_ty) = self_ty {
-                        if let Err(_) = self.at(&ObligationCause::dummy(), self.param_env)
-                            .sup(fty.inputs()[0], self_ty)
-                        {
+                        if self.at(&ObligationCause::dummy(), self.param_env)
+                               .sup(fty.inputs()[0], self_ty).is_err() {
```

Comment:
Nit: I'd prefer to see `is_err` on its own line here

Reply:
Sounds good, I'll update the commit soon.